### PR TITLE
Remove installer

### LINKED ISSUE #977: Add instance-store AMIs
We should build and publish S3-backed AMIs, allowing EC2 instances to be booted without EBS, which removes a failure mode for running instances. Packer supports this with the `amazon-instance` builder.

### LINKED ISSUE #1113: installer: Start bootstrapping as soon as AWS instances are up
We don't need to wait for the CloudFormation stack to come all the way up before we start bootstrapping. We just need to wait for the `CREATE_COMPLETE` events for each of the instances. The DNS propagation step takes a while so it makes sense to start bootstrapping as soon as we can.

### LINKED ISSUE #1246: Add flag to override domain for cluster install on aws
Would like a way to supply domain in flynn install aws to change out flynnhub domain name with custom domain such as cloud.example.com.

### LINKED ISSUE #1424: Route53 wildcard should be A alias
Instead of a CNAME, the installer wildcard Route53 record should be an A record with an ALIAS (health evaluation on) pointing at the domain, this avoids a potential second DNS lookup caused by the CNAME.

### LINKED ISSUE #1523: Install: Configurable VM disk size
**Updated issue**

Host disk sizes should be configurable, eg by doing:

```
$ VM_DISK_SIZE=120 flynn install
$ flynn install --vm-disk-size=120
```

---

Deploying often to Flynn causes the host to run out of disk space. As @lmars mentioned, there is a more complex issue regarding ZFS here.

Since this issue makes it impossible to use a cluster for more than a few days, I propose to at least get more time by increasing the disk size, while the bigger problem get's tackled with more time.

The idea is to make the VM disk size configurable by either setting an env var, a parameter or choose a form option from the install ui.

```
$ VM_DISK_SIZE=120 flynn install
$ flynn install --vm-disk-size=120
```

### LINKED ISSUE #1624: installer: Document development flow
The development flow for the installer should be documented in https://flynn.io/docs

When working on the installer app it's much faster to run it on it's own rather than running make and bootstrapping after each change.

Steps for getting it up and running:

1) Create or replace `installer/bindata.go` with the contents of https://gist.github.com/jvatic/9aa507d330b4c80e644a if working on the front-end (you'll need to repeat this if/after running `make` in the VM).

2) `cd installer/app && ENVIRONMENT=development go run compiler.go` (repeat after each modification to a JavaScript or CSS file).

3) From `{project root}/installer`, run `go build ../cli && ./cli install` (repeat after each modification to a go file, be sure not to commit the `cli` binary (you can delete it once the installer is running)).

### LINKED ISSUE #1719: Build Vagrant boxes for VMware and QEMU
The current VM release pipeline only builds Virtualbox and AWS Vagrant boxes.
I propose this is expanded to cover additionally VMware and QEMU.
This can be accomplished by modifying the way the build process works.

Specifically the changes need are:
- Install VMware Player on the build server
- Install QEMU on build server
- Change build process to download the Ubuntu Cloud OVA
- Use ovftool to create a VMX file for use with a new Packer builder that will go from VMX -> box
- Convert the VMDK inside the OVA to QCOW2 and use it with a new QEMU builder to produce the libvirt compatible box.

The proposed build process looks like this:

```
+-------+      +-------+      +----------+
|  OVA  +----->+  VMX  +----->+  VMware  |
+---+---+      +-------+      +----------+
    |                                     
    |          +-------+      +----------+
    +--------->+ QCOW2 +----->+   QEMU   |
    |          +-------+      +----------+
    |                                     
    |                         +----------+
    +------------------------>+VirtualBox|
                              +----------+
```

### LINKED ISSUE #1823: Use TUF verification when using install script from installer
Since the CLI has access to a working TUF repo, it can use that repo to securely verify the install script.

### LINKED ISSUE #1831: Use SSH host keys from discovery service
The installer should pull the SSH host keys from the discovery service and use them to authenticate the connection when we use cloudinit.

### LINKED ISSUE #1854: installer: Prefix log messages with host ip when multiple targets

### LINKED ISSUE #1876: installer: Update to React v15

### LINKED ISSUE #2096: installer: Show progress bar instead of log output by default in UI
Show a progress bar by default with an option to view the log stream.

### LINKED ISSUE #2099: installer: Cluster in-place update
Add action for running `flynn-host update`.

### LINKED ISSUE #2100: installer: Install from existing cluster
This takes launching from a backup a step further; The UI should make it easy to launch a new cluster from any cluster the installer is aware of. This should take the form of a cluster picker as part of the launcher and a clone action under each cluster to pre-populate the launcher to match it's configuration and restore from it.

Depends on #2098.

### LINKED ISSUE #2167: Command to ssh into flynn cluster
It would be great to have an easy way to ssh into a flynn cluster. Right now, [the documented way](https://flynn.io/docs/faq/how-do-i-ssh-to-flynn-cluster-hosts) to do this is to find out the ip of my cluster and run `ssh -l root -i ~/.flynn/installer/keys/flynn <ip>`

Maybe something like `flynn cluster ssh <cluster-name>` would be easier?

### LINKED ISSUE #2226: installer: Remove cluster from flynnrc when deleting

### LINKED ISSUE #2227: installer: Improve key selection when launching cluster
- Allow setting default key per provider
- Try other local keys (must exist locally and on the provider) before prompting for new key name

### LINKED ISSUE #2237: installer: Allow adding more nodes to running cluster

### LINKED ISSUE #2263: installer: Improve logging
- [ ] Include user/host/port for errors relating to ssh connections (and include more detail in error if possible), #1854 is covered by this
- [ ] Save individual cluster logs in the installer dir (i.e. `.flynn/installer/logs` in addition to `.flynn/installer/keys`)
- [ ] Make logs more verbose

### LINKED ISSUE #2350: Expose Data volume size as a variable in the web UI
It's already a variable in the CloudFormation stack template: https://github.com/flynn/flynn/blob/master/installer/stack_template.go#L28-L37

### LINKED ISSUE #2617: installer: SSH installer should check installer keys dir

### LINKED ISSUE #2645: installer: Extract core functionality into pkg
Move logic for managing Flynn on various platforms into `pkg/installer`. This will help clean things up a bit and open up more use cases.

The pkg should handle all interactions with the platforms and output a stream of events the consumer can use to update database records, create log events, etc.

### LINKED ISSUE #2665: installer: Add domain management for running clusters
Add ability to migrate domain via the installer UI and also add/remove domains, e.g. to/from Route53 on an AWS install.

(Note: #1246 covers configuring a domain to use before launching the cluster.)

### LINKED ISSUE #2687: Proposal for self-scaling clusters
Proposal:

Clusters could be configured for zero or more platforms to allow scaling instances and updating DNS for configured routes accordingly.

DNS platforms:
- AWS
- Azure
- Flynnhub (for flynnhub.com domains)
- DNSimple?
- Namecheap?

Instance platforms (add/remove):
- AWS
- Azure
- DigitalOcean

Clusters launched via the installer would be configured automatically by default and operators could configure them manually via the cli or dashboard.

Thoughts?

### LINKED ISSUE #2754: Clarify storage section of the operating docs
It should mention that `flynn install` on EC2 does the right thing and expand on the need for separate drive(s).

### LINKED ISSUE #2832: Please allow selection of an existing VPC and subnets when installing to AWS
I would like to use flynn as the deployment platform for an app that uses other pre-existing AWS services. During flynn's installation phase, I would like the option to provide an existing VPC to install the platform to in a fashion that is similar to [deis](http://docs.deis.io/en/latest/installing_deis/aws/#launch-into-an-existing-vpc). 

`VPC_ID` - The ID of an existing VPC to install into
`VPC_SUBNETS` - A comma-separated list of subnet IDs
`VPC_ZONES` - A comma-separated list of regions and availability zones matching the order of provided subnets

### LINKED ISSUE #2842: Allow user-defined AWS IAM roles to be assigned to nodes during cluster creation
I'm using DataDog to monitor our cluster, and in order to get custom tags pushed to DataDog from the agent config running on the Flynn nodes, the nodes have to be associated with an IAM role. The only problem is IAM roles cannot be associated with existing EC2 instances, only newly created instances.

See "Can I associate an IAM role with an already running EC2 instance?" here:
https://aws.amazon.com/iam/faqs/

Also this AWS forum thread:
https://forums.aws.amazon.com/thread.jspa?threadID=97487

### LINKED ISSUE #2893: Flynn install command should allow cluster creation from commandline
I currently have to run `flynn install` and then create a cluster by clicking in the web UI. Alternatively, I'd like to be able to create an instance by doing something like:

```
>  flynn install -n $cluster_name -c $credentials --aws_region $region --aws_instance_type $type --instance_count 3
```

Something like the above would allow me to automate most of the process for bringing up a new cluster.

### LINKED ISSUE #2905: Feature request: ability to restore cluster from S3/URL using installer
This is a follow up from my previous issue, and I don't know how complicated it would be, but it would be really nice if the installer were smart enough to be able to bootstrap a cluster by downloading a backup from a URL or service like S3 instead of uploading/downloading from local machine.

We're planning on storing regular backups on S3, but I think you can just send S3 an HTTP-basic auth URL to download files, so just having a "URL" field would be super nice.

If you're downloading and uploading large files over Mexican internet, you're not gonna have a fun time. ;-)

### LINKED ISSUE #2924: installer: Add S3 blobstore option for AWS clusters
See #2922

### LINKED ISSUE #2938: Allow user to specify cluster name during creation
Allow the user to specify the name makes it easier to work with Flynn post-creation, especially if you have multiple clusters. Just one example:

```
> flynn -c prod apps

vs.

> flynn -c flynn-1234567 apps
```

I know I can create an environment variable in .bashrc, which is what I'm currently doing. But that doesn't help other engineers.

This change would also reduce the opportunities for mistakes. Is `flynn-1234567` production or is `flynn-1234576`?

### LINKED ISSUE #3020: Require typed confirmation for installer cluster delete
The deletion confirmation should require typing the cluster name before proceeding to prevent accidental deletion.

### LINKED ISSUE #3190: feature request: allow specification of Availability Zone in AWS installer
Because Reserved Instances are tied to a specific AZ, it'd be nice to be able to use the Flynn Installer to spin up installations in specific AZs.

Proposed functionality..

On the installer UI, next to or below the Region selection, add another selection with region-specific AZs.

This second selector may default to "Any".

Its options should include the AZs relative to the currently selected Region. So, if a user chooses US East, the AZs should be:
- Any
- us-east-1a
- us-east-1b
- us-east-1c
- us-east-1d
- us-east-1e

(I'm not sure if you can pull the available AZs dynamically from the AWS API, but I'm guessing "yes, you can")

### LINKED ISSUE #3195: SSH installation failing with both password and key-based auth
Feel free to test with this host as an example.

**IP:** 173.255.218.11
**User:** root
**Pass:** TAeXkwfsYxRFREcmeNdbAZrj42yNmu

### LINKED ISSUE #3278: Allow changing cluster name
Once a cluster has been created, is there a way to change its name?
By default flynn generates a random name and but I don't see any field to set a name for the cluster.

### LINKED ISSUE #3341: Installer UX issues
Okay, first of all, your installer is great.  Seriously it's awesome.  You know, the one that fully automates the setup even for bare metal?

But it does some things that waste serious time:
- If I get a password wrong, please don't make me re-enter everything
- If my machine is being obnoxious because the host id has changed on my machines, please don't make me re-enter everything
- In fact, why not let me enter the info about the cluster, and then if it is interrupted, I can go back to the start with the IPs and stuff already entered.  
- If I'm using a password, why not let me enter it from the start?

Also, I got an error on installing.  I'm trying it once more, and if that doesn't work I'll let you know in another issue.  

Anyway-- Flynn is incredible and quite promising!  Thanks for all your hard work!

### LINKED ISSUE #3385: installer: Logging should be using ISO 8601 or at least have an option to enable it
In the logs shown by flynn the dates are shown in a way that really only relates to some countries as most use DD-MM vs the current MM-DD. I suggest instead of using that you move to the standard ISO 8601 and use UTC time.

### LINKED ISSUE #3477: Investigate more efficient method of obtaining Azure credentials
Either through wrapping the Azure CLI or some other method.

### LINKED ISSUE #3504: UI is not listing clusters
I installed Flynn on Fedora 24 as root by

```
L=/usr/local/bin/flynn && curl -sSL -A "`uname -sp`" https://dl.flynn.io/cli | zcat >$L && chmod +x $L
```

I created and added clusters which are listed in the command line:

```
flynn cluster
NAME              CONTROLLER URL                        GIT URL                        DOCKER URL
dev               https://controller.[xxx].flynnhub.com  https://git.[xxx].flynnhub.com  (none)
flynn-1476460741  https://controller.[xxx].flynnhub.com  https://git.[xxx].flynnhub.com  (none)  (default)
```

But none of these clusters appear on the left side of the graphical interface which pops up after

```
flynn install
```

This list is probably important as it is the only way I found how to destroy clusters. Right now I need to ask colleagues (who use macbooks) to remove them for me. Is this a bug or change in interface?

Would it be possible to add a cluster deletion option to CLI?

### LINKED ISSUE #3538: AWS install stops with Authentication failed - policy incorrect?
using `flynn install` for an AWS cluster, have tried 3 sets of credentials on 2 different accounts. Have set up with the managed policies for AmazonEC2FullAccess, AmazonS3FullAccess and a custom policy for Cloudformation (AWS doesn't have one pre-built) that looks like this:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1476880971000",
            "Effect": "Allow",
            "Action": [
                "cloudformation:*"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}
```

the logs just look like this:

```
$ flynn install
Open http://localhost:49637 in your browser to continue.
INFO[10-19|09:30:43] Event: new_cluster:
INFO[10-19|09:30:43] Event: log: Using saved key pair (flynn)
INFO[10-19|09:30:43] Event: log: Allocating domain
INFO[10-19|09:30:43] Event: log: Fetching image manifest
INFO[10-19|09:30:44] Event: log: Generating start script
INFO[10-19|09:30:44] Event: log: Creating stack
INFO[10-19|09:30:53] Event: prompt:
```

The UI shows "Authentication failed, please select a new credential to continue". Not sure what operation is being performed, but I think the credentials are correct, because if I detach the Cloudformation policy, the logs show:

`Error: User: arn:aws:iam::588275505772:user/flynn-test is not authorized to perform: cloudformation:CreateStack on resource: arn:aws:cloudformation:us-east-1:588275505772:stack/flynn-1476884287/*`

It's only when I actually have the Cloudformation policy attached that I get the "Authentication failed" result.

Not sure what policies people are using to install on AWS, but if it's full admin access, we won't be able to use that. So, just wondering what the setup should be.

### LINKED ISSUE #3617: Browser installer - Install dependencies before installing anything
Could the `flynn install` command first check for and install dependencies if needed, bailing out before making any chances if it can't pre-setup the system?

In some cases (in mine specifically a dedicated server at online.net) Ubuntu 14.04 comes as an extremely minimal install so doesn't include curl by default. Installing curl is an easy enough thing to do but it does make me wonder if other things are missing.

It also causes the message "If Flynn is already installed, it will be stopped and removed along with all data before installing the latest version. Would you like to proceed?" which only seems to ask for one server in a multi-server setup - eventually the easiest fix is to reinstall the OS on everything and start over hoping you don't miss something in take two

### LINKED ISSUE #3802: Cluster setup from web installer may fail related to netfilter-persistent
Attempting to set up a 3x 2 GB / 40 GB cluster on DigitalOcean via the web installer failed multiple times before successfully completing on the third of fourth try.  I don't have complete logs from the failed startup, so I'll do the best I can to describe what I remember seeing last night.

First off, I know that the logs included the lines:
'netfilter-persistent: command not found' and
'run-parts: executing /usr/share/netfilter-persistent/plugins.d/15-ip4tables save'

The failures appeared to hang in different places, and I don't believe any of them stopped immediately upon hitting the initial issue with netfilter-persistent. There would be a message about a missing dependency or service setup failure around netfilter-persistent, and then the installer would print some lines related to setting it up (looked like a delayed or retry attempt), and finally the install would get stuck. I waited about 5-10 minutes each time before killing the install and starting over.

There may have been some interplay between previously failed (but not fully cleaned up clusters) and the new deployment attempts, but it seems pretty unlikely. The only reason I'm mentioning it is because I made sure to fully clean up the previous failed clusters before the last attempt finally completed successfully.

### LINKED ISSUE #3821: Installer references Ubuntu 14.04
I'm installing on 16.04 via SSH

### LINKED ISSUE #3827: AWS region ap-south-1(Mumbai) not available
While creating the cluster using AWS option the region ap-south-1(Mumbai) is not listed in available options

### LINKED ISSUE #3868: Add Google Cloud Platform installer
Specifically, Google Compute Engine. I'm surprised this wasn't available.

Is there some limitation I'm unaware of, and do you accept pull requests?

### LINKED ISSUE #4052: install on digitalocean is still not working with v20170425.0
After closing of #4046 I tried to install on digitalocean (Frankfurt 4GB) right now.  It still does not work:

```INFO[04-25|11:00:50] Event: new_cluster:                                                                                   
INFO[04-25|11:00:51] Event: log: Using saved key pair (flynn)                                                              
INFO[04-25|11:00:51] Event: log: Allocating domain                 
INFO[04-25|11:00:51] Event: log: Configuring DNS         
INFO[04-25|11:00:52] Event: log: Creating droplet flynn-1493110850
INFO[04-25|11:01:15] Event: log: Fetching droplets for flynn-1493110850
INFO[04-25|11:01:16] Event: log: Configuring domain                             
INFO[04-25|11:01:17] Event: log: Installing flynn            
INFO[04-25|11:01:18] Event: log: Waiting for ssh on 46.101.115.54                      
INFO[04-25|11:01:30] Event: log: Configuring iptables firewall on 46.101.115.54
INFO[04-25|11:01:30] Event: log: Running `export DEBIAN_FRONTEND=noninteractive
apt-get update                                     
apt-get install -y iptables-persistent            
iptables -F INPUT                        
iptables -A INPUT -s 46.101.115.54 -j ACCEPT

iptables -A INPUT -i eth0 -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
iptables -A INPUT -i eth0 -p tcp --dport 80 -j ACCEPT
iptables -A INPUT -i eth0 -p tcp --dport 443 -j ACCEPT
iptables -A INPUT -i eth0 -p tcp --dport 22 -j ACCEPT
iptables -A INPUT -i eth0 -p icmp --icmp-type echo-request -j ACCEPT
iptables -A INPUT -i eth0 -j DROP
netfilter-persistent save
` on 46.101.115.54
INFO[04-25|11:01:31] Event: log: Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
INFO[04-25|11:01:31] Event: log: Get:2 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]
INFO[04-25|11:01:31] Event: log: Get:3 http://security.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
INFO[04-25|11:01:31] Event: log: Get:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [102 kB]
INFO[04-25|11:01:32] Event: log: Get:5 http://archive.ubuntu.com/ubuntu xenial/universe amd64 Packages [7,532 kB]
INFO[04-25|11:01:32] Event: log: Get:6 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages [255 kB]
INFO[04-25|11:01:32] Event: log: Get:7 http://security.ubuntu.com/ubuntu xenial-security/main Translation-en [108 kB]
INFO[04-25|11:01:32] Event: log: Get:8 http://archive.ubuntu.com/ubuntu xenial/universe Translation-en [4,354 kB]
INFO[04-25|11:01:32] Event: log: Get:9 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [109 kB]
INFO[04-25|11:01:32] Event: log: Get:10 http://security.ubuntu.com/ubuntu xenial-security/universe Translation-en [55.8 kB]
INFO[04-25|11:01:32] Event: log: Get:11 http://security.ubuntu.com/ubuntu xenial-security/multiverse amd64 Packages [2,748 B]
INFO[04-25|11:01:32] Event: log: Get:12 http://security.ubuntu.com/ubuntu xenial-security/multiverse Translation-en [1,232 B]
INFO[04-25|11:01:33] Event: log: Get:13 http://archive.ubuntu.com/ubuntu xenial/multiverse amd64 Packages [144 kB]
INFO[04-25|11:01:33] Event: log: Get:14 http://archive.ubuntu.com/ubuntu xenial/multiverse Translation-en [106 kB]
INFO[04-25|11:01:33] Event: log: Get:15 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [520 kB]
INFO[04-25|11:01:33] Event: log: Get:16 http://archive.ubuntu.com/ubuntu xenial-updates/main Translation-en [210 kB]
INFO[04-25|11:01:33] Event: log: Get:17 http://archive.ubuntu.com/ubuntu xenial-updates/restricted amd64 Packages [7,776 B]
INFO[04-25|11:01:33] Event: log: Get:18 http://archive.ubuntu.com/ubuntu xenial-updates/restricted Translation-en [2,548 B]
INFO[04-25|11:01:33] Event: log: Get:19 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [459 kB]
INFO[04-25|11:01:33] Event: log: Get:20 http://archive.ubuntu.com/ubuntu xenial-updates/universe Translation-en [176 kB]
INFO[04-25|11:01:33] Event: log: Get:21 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 Packages [8,916 B]
INFO[04-25|11:01:33] Event: log: Get:22 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse Translation-en [4,136 B]
INFO[04-25|11:01:33] Event: log: Get:23 http://archive.ubuntu.com/ubuntu xenial-backports/main amd64 Packages [4,684 B]
INFO[04-25|11:01:33] Event: log: Get:24 http://archive.ubuntu.com/ubuntu xenial-backports/main Translation-en [3,216 B]
INFO[04-25|11:01:33] Event: log: Get:25 http://archive.ubuntu.com/ubuntu xenial-backports/universe amd64 Packages [2,512 B]
INFO[04-25|11:01:33] Event: log: Get:26 http://archive.ubuntu.com/ubuntu xenial-backports/universe Translation-en [1,216 B]
INFO[04-25|11:01:37] Event: log: Fetched 14.4 MB in 5s (2,869 kB/s)
INFO[04-25|11:01:38] Event: log: Reading package lists...
INFO[04-25|11:01:38] Event: log: Reading package lists...
INFO[04-25|11:01:38] Event: log: Building dependency tree...
INFO[04-25|11:01:38] Event: log: E: Unable to locate package iptables-persistent
INFO[04-25|11:01:38] Event: log: Reading state information...
INFO[04-25|11:01:38] Event: log: bash: line 13: netfilter-persistent: command not found
EROR[04-25|11:01:38] Error: Process exited with status 127
INFO[04-25|11:01:38] Event: cluster_state: error
```

to be sure:
```
[martin@martin-arch ~]$ flynn --version
v20170425.0
```

Thanks.

### PR DESCRIPTION
**Pull request by titanous**
This removes the `flynn install` command and the EC2 builder that it depended on. The installer has proven to be a bug magnet and the Flynn team doesn't have the time to maintain it and deal with the support burden that it creates.

An unsupported and unmaintained snapshot of installer binaries at the time of deprecation is available at https://dl.flynn.io/flynn-install-deprecated.tar.gz

Closes #977
Closes #1113
Closes #1246
Closes #1424
Closes #1523
Closes #1719
Closes #1624
Closes #1823
Closes #1831
Closes #1854
Closes #1876
Closes #2096
Closes #2099
Closes #2100
Closes #2167
Closes #2226
Closes #2227
Closes #2237
Closes #2263
Closes #2350
Closes #2617
Closes #2645
Closes #2665
Closes #2687
Closes #2754
Closes #2832
Closes #2842
Closes #2893
Closes #2905
Closes #2924
Closes #2938
Closes #3020
Closes #3123
Closes #3190
Closes #3195
Closes #3278
Closes #3341
Closes #3385
Closes #3477
Closes #3504
Closes #3538
Closes #3617
Closes #3802
Closes #3821
Closes #3827
Closes #3868
Closes #3959
Closes #3973
Closes #4052

### CODE REVIEW

File: docs/content/installation.html.md [hunk 1]
```diff
@@ -10,12 +10,8 @@ There are several ways to install Flynn.
 To install or configure the Flynn client command line interface, [visit the CLI
 docs](/docs/cli).
 
-If you'd like to install Flynn on cloud servers or via SSH using an easy and
-secure local graphical interface, [use the cloud
-installer](/docs/installation/cloud).
+If you want to install a Flynn cluster, follow the [installation
+guid](/docs/installation/manual) guide.
```
Comment (lmars) [inline docs/content/installation.html.md]: s/guid/guide and drop the duplicate "guide" after the link.

Comment (titanous) [inline docs/content/installation.html.md]: Fixed.

**Comment by arekkas** [comment]
Damn

**Comment by titanous** [comment]
Here's some additional background for this change:

The installer started as an experiment with the intent of launching demo clusters. We added some features to it, but never got it to the point where we felt it could run and manage reliable production clusters with best practices out of the box. Due to the surface area of cloud provisioning, we ran into a variety of bugs, and don't have the team bandwidth to fix them and redo the architecture to support production clusters. These issues have kept building up, and instead of continuing to ship a subpar tool that we don't currently have the resources to maintain, we decided to remove it.

Ease of use is one of the primary goals of Flynn, and we continue to look for the best ways to help users install and manage Flynn clusters.

**Comment by WriteCodeEveryday** [comment]
@titanous @lmars Is there any chance we could discuss keeping the SSH installer intact as a middle ground (I know it's merged and all, but this one came out of nowhere)? 

When doing demos for companies looking for a easier way to manage their online apps, I find the installer a good way to show how Flynn installs without going into the 'terminal'. Honestly, this was a place you guys really shone above Deis and Kubernetes. I've never had an issue with it (the cloud ones were awesome and I rarely had any issues with them as well). I just don't want to have to choose between updating my cli version or using the installer, because I am 100% sure all of you are planning on continuing to develop and make Flynn awesome. 

Maybe it should have deprecated with warning (in the installer html page in giant bold red letters) but without removing and just closing the Github installer issues in the future. I really don't want to have to fork this and manually update and build it from now.

**Comment by titanous** [comment]
We split the deprecated installer out into a separate binary that will not be updated or supported and is available here: https://dl.flynn.io/flynn-install-deprecated.tar.gz

Note that we are no longer building AMIs, so the AWS installer will not necessarily install the latest version of Flynn.

**Pull request merged**